tests/unittests/tests-bhp_msg/tests-bhp_msg.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2022 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "bhp/msg.h"
+#include "embUnit.h"
+
+#include "unittests-constants.h"
+#include "tests-bhp_msg.h"
+
+static bhp_msg_t bhp_msg;
+static int canary;
+static int *ctx;
+
+static char _stack[THREAD_STACKSIZE_DEFAULT];
+
+static void bhp_handler(void *arg)
+{
+    TEST_ASSERT(arg == ctx);
+    canary = TEST_INT;
+}
+
+static void set_up(void)
+{
+    memset(&bhp_msg, '\0', sizeof(bhp_msg));
+    canary = 0;
+    bhp_msg_init(&bhp_msg, bhp_handler, ctx);
+}
+
+static void test_bhp_msg__init(void)
+{
+    TEST_ASSERT(bhp_msg.pid == KERNEL_PID_UNDEF);
+    TEST_ASSERT(bhp_msg.msg.type == BHP_MSG_BH_REQUEST);
+    TEST_ASSERT(bhp_msg.bhp.irq_handler == bhp_handler);
+    TEST_ASSERT(bhp_msg.bhp.ctx == ctx);
+}
+
+static void *_event_loop(void *arg)
+{
+    (void) arg;
+    while (1) {
+        msg_t msg;
+        msg_receive(&msg);
+        if (msg.type == BHP_MSG_BH_REQUEST) {
+            bhp_irq_handler(msg.content.ptr);
```

Comment:
Maybe add a helper function

```C
static inline void bhp_msg_handle(msg_t *msg)
{
    assert(msg->type == BHP_MSG_BH_REQUEST);
    bhp_irq_handler(msg->content.ptr);
}
```

so it's more obvious what to do with the msg

Reply:
Hm I suppose that's a no?

Reply:
done. I amended it directly.